pkg/compare/container_test.go

```diff
@@ -0,0 +1,326 @@
+package compare
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"testing"
+)
+
+func TestIsContainer(t *testing.T) {
+	tests := []struct {
+		input    string
+		expected bool
+	}{
+		{"container://my-image:latest:/metadata.yaml", true},
+		{"file://local/path/to/file.yaml", false},
+		{"http://example.com/data.yaml", false},
+		{"randomstringwithoutprefix", false},
+	}
+
+	for _, test := range tests {
+		result := isContainer(test.input)
+		if result != test.expected {
+			t.Errorf("For input '%s', expected %v, got %v", test.input, test.expected, result)
+		}
+	}
+}
+
+func TestParsePath(t *testing.T) {
+	tests := []struct {
+		input         string
+		expectedImage string
+		expectedPath  string
+		expectError   bool
+	}{
+		{"container://my-image:latest:/metadata.yaml", "my-image:latest", "/metadata.yaml", false},
+		{"container://repo/image:tag:/data.yaml", "repo/image:tag", "/data.yaml", false},
+		{"container://repo/image:::tag::::/data.yaml", "repo/image:tag", "/data.yaml", false},
+		{"container://wrongformat", "", "", true},
+		{":missingprefix:/file.yaml", "", "", true},
+		{"container://only:two", "", "", true},
```

Comment:
As above, try to think like a user who has a 2-year-old helping them type commands in ways that would potentially confuse the parser:
- `"container://"`
- `"container:/"`
- `"container:"`
- `"container"`
- `""`